File: Kadane's.py
def MaxSubarraySum(a)->int:
    currSum = 0
    maxSum = 0
    for i in range(len(a)):
        currSum += a[i]
        maxSum = max(maxSum , currSum)
        if(currSum < 0):
            currSum = 0
    return maxSum

if __name__ == "__main__":
    n = int(input("Enter the number of elements : "))
    arr = list(map(int, input(f"Enter {n} array elements: ").split()))
    print(MaxSubarraySum(arr))

File: Kadane's.cpp
#include<iostream>
#include<vector>
using namespace std;
int MaximumSubarraySum(vector<int> &arr)
{
	int currSum = 0;
	int maxSum = 0;
	for(int i = 0 ; i < arr.size() ; i++)
	{
		currSum += arr[i];
		maxSum = max(maxSum , currSum);
		if(currSum < 0)
		{
			currSum = 0;
		}
	}
	return maxSum;
}
int main()
{
	
	int n;
	vector<int> arr(n);
	cout<<"Enter the number of elements : ";
	cin>>n;
	
	cout<<"Enter the array elements : ";
	for(int i = 0 ; i < n ; i++)
	{
		cin>>arr[i];
	}
	cout<<MaximumSubarraySum(arr)<<endl;
	
	return 0;
}
